### ISSUE TITLE
Minifier dumps checkpoints which don't actually reproduce the error

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#93694]
Steps to reproduce:
1. Checkout 07d794094bd4eaf1ad946429044f1c742f278d6b of pytorch (at time of writing, this is tip of symbolic-shapes branch)
2. Checkout e50d6695990fda7a0dfde460c6c6a2f93e1c8d1d of torchdynamo (at time of writing, this is tip of accuracy-minifier branch)
3. Intentionally introduce a failure in pytorch

```
diff --git a/torch/_decomp/__init__.py b/torch/_decomp/__init__.py
index 9512b0f38a..7f3d29290e 100644
--- a/torch/_decomp/__init__.py
+++ b/torch/_decomp/__init__.py
@@ -100,7 +100,6 @@ def register_decomposition(aten_op, registry=None, *, disable_meta: bool = False
                 if op_overload in registry:
                     raise RuntimeError(f"duplicate registrations for {op_overload}")
                 registry[op_overload] = fn
-                op_overload.py_impl(torch._C.DispatchKey.Meta)(fn)
                 # TODO: factor this logic into OpOverload or Library API
                 name = op_overload._schema.name
                 if op_overload._schema.overload_name:
```

4. Trigger the error and try to minify it `(cd ../torchdynamo && TORCHDYNAMO_REPRO_AFTER=dynamo TORCH_SHOW_CPP_STACKTRACES=1 AOT_FX_GRAPHS_JOINT=1 TORCHDYNAMO_DYNAMIC_SHAPES=1 AOT_DYNAMIC_SHAPES=1 time python benchmarks/torchbench.py --only BERT_pytorch --accuracy --backend aot_eager --training)`

This produces this checkpoint /tmp/minifier_ezyang/checkpoints/2.py

```
import torch
import torchdynamo
from torch import tensor, device
import torch.fx as fx
from torchdynamo.testing import rand_strided
from math import inf
from torchdynamo.debug_utils import run_fwd_maybe_bwd


from torch.nn import *
class Repro(torch.nn.Module):
    def __init__(self):
        super().__init__()

    
    
    def forward(self, embedding):
        return (embedding,)
        

mod = Repro().cuda()
opt_mod = torchdynamo.optimize("aot_eager")(mod)


args = [((2, 128, 768), (98304, 768, 1), torch.float32, 'cuda', True)]
args = [rand_strided(sh, st, dt, dev).requires_grad_(rg) for (sh, st, dt, dev, rg) in args]


with torch.cuda.amp.autocast(enabled=False):
    ref = run_fwd_maybe_bwd(mod, args)
    res = run_fwd_maybe_bwd(opt_mod, args)
```

However this script runs without reproducing the error. It looks like the final repro.py DOES reproduce the error, so I am confused why there's a checkpoint that doesn't do it.

cc @soumith @msaroufim @wconstab @ngimel @bdhirsh @anijain2305
